bundles/net.tourbook.device.garmin.fit/src/net/tourbook/device/garmin/fit/listeners/MesgListener_Record.java

```diff
@@ -311,10 +310,14 @@ private void setRecord(final RecordMesg mesg) {
     */
    private void setRecord_DeveloperData(final RecordMesg mesg, final TimeData timeData) {
 
-      final boolean containsMultiplePowerDataSources =
-            StreamSupport.stream(mesg.getDeveloperFields().spliterator(), false)
-                  .anyMatch(developerField -> developerField.getName().equals(DEV_FIELD_NAME__POWER) &&
-                        developerField.getName().equals(DEV_FIELD_NAME__POWER));
+      int powerDataSources = 0;
```

Comment:
this should be moved below line 361 because it is used only there

Reply:
But I thought it would be better to have it out of the for loop so we don't recalculate it every time (more efficient). Thoughts?

Reply:
When entering the method, you can check if dev data are available

if (mesg.getDeveloperFields().size() == 0) {
   return
}

Reply:
Ok, this comment doesn't apply anymore since I have removed this code

Reply:
The following code is computed for every record in all fit files

![fit-record-listener](https://user-images.githubusercontent.com/1283445/63221013-e8679800-c192-11e9-92fc-ea20802ffc74.png)

When putting
```
if (mesg.getDeveloperFields().size() == 0) {
   return;
}
```
at the beginning of the method, any dev field computation would be not necessary

Reply:
Oh, got it. I misunderstood. I will fix it ASAP.

Reply:
Done (see upcoming PR)